app/logic/serviceLearningCoursesData.py

```diff
@@ -30,6 +36,35 @@ def withdrawProposal(courseID):
     CourseInstructor, Note"""
     course = Course.get(Course.id == courseID)
     questions = CourseQuestion.select().where(CourseQuestion.course == course)
-    notes = list(Note.select(Note.id).where(QuestionNote.question.in_(questions)).distinct().join(QuestionNote))
+    notes = list(Note.select(Note.id)
+                .join(QuestionNote)
+                .where(QuestionNote.question.in_(questions))
+                .distinct())
     course.delete_instance(recursive=True)
-    Note.delete().where(Note.id.in_(notes)).execute()
+    for note in notes:
+        note.delete_instance()
+
+def renewProposal(courseID, term):
+    course = Course.get(Course.id == courseID)
+    oldTerm = course.term
+    course.id = None
+    course.term = Term.get_by_id(term)
+    oldCourse = Course.get(courseName=course.courseName, term=oldTerm)
+    oldCourse.status=course.status
+    course.status = 3
+    course.save()
+    oldCourse.save()
+    questions = (CourseQuestion.select(CourseQuestion, Course)
+                    .join(Course)
+                    .where(CourseQuestion.course==oldCourse.id,
+                    Course.term==oldTerm,
+                    Course.courseName==course.courseName))
```

Comment:
I'm not seeing the comment I thought I made on this section.The variables and assignments are a bit out of whack at the beginning of the function. You can get the old course object and the course object at the beginning and then use those in the rest of the function.